domains/client/domain-operator/src/domain_block_processor.rs

```diff
@@ -773,6 +773,20 @@ where
                     execution_receipt.hash(),
                     receipt_mismatch_info,
                 ));
+
+                continue;
+            }
+
+            if execution_receipt.domain_block_extrinsic_root
+                != local_receipt.domain_block_extrinsic_root
+            {
+                bad_receipts_to_write.push((
+                    execution_receipt.consensus_block_number,
+                    execution_receipt.hash(),
+                    ReceiptMismatchInfo::DomainExtrinsicsRoot {
+                        consensus_block_hash,
+                    },
```

Comment:
We need consensus block number since aux storage is fork aware and indexes number to block hashes. 

> This whole tuple looks strange: we have consensus block number, then has of ER, but then mismatch info enum just has consensus block hash, which is ignored below. There should be a more consistent way to do this.

I agree. We can make it consistent but right now we are re-using some data types from Fraud proof V1 and there is a TODO to clean it up as we do not know how Fraud proof V2 would look like right now